src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -185,43 +190,46 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
-            {
-                if (owner->GetTypeId() == TYPEID_UNIT)
-                {
-                    owner->ToCreature()->SetCannotReachTarget(false);
-                }
-
-                lastPathingFailMSTime = World::GetGameTimeMS();
-                owner->m_targetsNotAcceptable[i_target->GetGUID()] = MMapTargetData(sWorld->GetGameTime() + DISALLOW_TIME_AFTER_FAIL, owner, i_target.getTarget());
-                return;
-            }
-            else
-            {
-                owner->m_targetsNotAcceptable.erase(i_target->GetGUID());
-                owner->AddUnitState(UNIT_STATE_CHASE);
-
-                init.MovebyPath(i_path->GetPath());
-                if (i_angle == 0.f)
-                    init.SetFacing(i_target.getTarget());
-                init.SetWalk(((D*)this)->EnableWalking());
-                init.Launch();
-                return;
-            }
-        }
-        else
-        {
-            // evade first
-            if (owner->GetTypeId() == TYPEID_UNIT)
-            {
-                owner->ToCreature()->SetCannotReachTarget(true);
-            }
-            // then use normal MoveTo - if we have to
-        }
+         bool success = i_path->CalculatePath(x, y, z, forceDest);
+         if (!success)
+         {
+             if (owner->GetTypeId() == TYPEID_UNIT)
+             {
+                 owner->ToCreature()->SetCannotReachTarget(true);
+             }
+
+             owner->StopMoving();
+             return;
+         }
+
+         float maxDist = owner->GetMeleeRange(i_target.getTarget());
+         if (!forceDest && (i_path->GetPathType() & PATHFIND_NOPATH || (!i_offset && !isPlayerPet && i_target->GetExactDistSq(i_path->GetActualEndPosition().x, i_path->GetActualEndPosition().y, i_path->GetActualEndPosition().z) > maxDist * maxDist)))
+         {
+             if (owner->GetTypeId() == TYPEID_UNIT)
+             {
+                 owner->ToCreature()->SetCannotReachTarget(false);
+             }
+
+             lastPathingFailMSTime = World::GetGameTimeMS();
+             owner->m_targetsNotAcceptable[i_target->GetGUID()] = MMapTargetData(sWorld->GetGameTime() + DISALLOW_TIME_AFTER_FAIL, owner, i_target.getTarget());
+             return;
+         }
+
+         owner->m_targetsNotAcceptable.erase(i_target->GetGUID());
+         owner->AddUnitState(UNIT_STATE_CHASE);
+
+         init.MovebyPath(i_path->GetPath());
+         if (i_angle == 0.f)
+             init.SetFacing(i_target.getTarget());
+         init.SetWalk(((D*)this)->EnableWalking());
+         init.Launch();
+         return;
+    }
+
+    const Unit* unit = owner->ToUnit();
+    if (unit && !unit->GetMap()->CanReachPositionAndGetCoords(unit, x, y, z, true, true, false)) {
+        if (!forceDest)
```

Comment:
why not moving this condition in the upper `if` ?

Reply:
Also bracket on the over should be on a new line